runtime/compiler/env/VMJ9Server.cpp

```diff
@@ -1062,6 +1062,28 @@ TR_J9ServerVM::sameClassLoaders(TR_OpaqueClassBlock * class1, TR_OpaqueClassBloc
 bool
 TR_J9ServerVM::isUnloadAssumptionRequired(TR_OpaqueClassBlock *clazz, TR_ResolvedMethod *methodBeingCompiled)
    {
+   TR_OpaqueClassBlock * classOfMethod = static_cast<TR_ResolvedJ9JITaaSServerMethod *>(methodBeingCompiled)->classOfMethod();
+      {
```

Comment:
Let's optimize this code: if clazz==classOfMethod then we can return false immediately; no need to search the hashtables. Next you can do a search for `clazz`. With it you can evaluate whether getClassLoader(clazz)==getSystemClassLoader as well as isAnonymousClass(clazz). If you cannot answer false then you must search for classOfMethod